### ISSUE TITLE
[$1000] Share code - Light background does not appear below the avatar image (reported by @alitoshmatov)

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#20509]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/19883

## Action Performed:
1. Open the App and upload a profile picture that has a transparent background.
2. Go to Settings -> Share code

## Expected Result:
Verify that a light green background appears below the avatar image.

## Actual Result:
A light green background does not appear below the avatar image.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.26.1

**Reproducible in staging?:** yes

**Reproducible in production?:** yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/3af307e6-84e2-41dd-9cec-3f67d1abb002

**Expensify/Expensify Issue URL:**

**Issue reported by:** @alitoshmatov  /  Applause - Internal Team

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1686247175943009

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013a3d4105d7772fca</li>
        <li>Upwork Job ID: 1668408135973871616</li>
        <li>Last Price Increase: 2023-06-13</li>
    </ul>
</details>

**Comment by Nikhil-Vats** [comment]
I think this is a duplicate of https://github.com/Expensify/App/issues/19569. @allroundexperts can you confirm?

**Comment by s-alves10** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Light background doesn't appear below the avatar image

### What is the root cause of that problem?

We don't pass `logoBackgroundColor` props here so the default value(defaultTheme.icon) is used.

https://github.com/Expensify/App/blob/4dc58e7e53fea05459307eb18ff5b1213ce87eaa/src/components/QRCode/index.js#L62-L73

https://github.com/Expensify/App/blob/4dc58e7e53fea05459307eb18ff5b1213ce87eaa/src/components/QRCode/index.js#L53

### What changes do you think we should make in order to solve the problem?

Set the default `logoBackgroundColor` to `defaultTheme.highlightBG`

```
    logoBackgroundColor: defaultTheme.highlightBG
```

<details>
    <summary>Result</summary>

</details>



### What alternative solutions did you explore? (Optional)

**Comment by alitoshmatov** [comment]
I have reported this here - https://expensify.slack.com/archives/C049HHMV9SM/p1686247175943009

There are two problems
1. When png uploaded, backend is converting it to jpeg and it is getting default dark background
2. https://github.com/Expensify/App/pull/19883 produced the regression where every logo in qr code have grey background

**Comment by strepanier03** [comment]
@alitoshmatov - Is this being fixed with the [PR](https://github.com/Expensify/App/pull/19883) you mentioned? Do you recommend we put this GH on hold for now?

**Comment by alitoshmatov** [comment]
@strepanier03 No this PR is the one causing the regression, right now there is no PR to fix this yet. I think someone should confirm that this is indeed regression of https://github.com/Expensify/App/pull/19883 and state next steps.

**Comment by strepanier03** [comment]
Got it, thanks @alitoshmatov!

**Comment by mollfpr** [comment]
@strepanier03 @aldo-expensify I'm unsure if this a regression from https://github.com/Expensify/App/pull/19883 as it is [expected](https://github.com/Expensify/App/issues/19569#issuecomment-1570140175) before @allroundexperts put up the PR.

Although, the result is not correct as expected in other PR that hit the production first https://github.com/Expensify/App/pull/19984#issuecomment-1573745740

**Comment by strepanier03** [comment]
@aldo-expensify - Can you please help up determine the regression here and next steps? I am heading out for sabbatical this evening so I will be reassigning this to another BZ momentarily.

**Comment by strepanier03** [comment]
@trjExpensify - Sorry to give you another but I've got a lot of BZ issues to hand off 😬. 

I bumped the eng here to help figure out the regression and then this will need some pressure to keep moving forward. For today there is no immediate action required from you. 

Thank you for wrapping this up 🙌

**Comment by aldo-expensify** [comment]
@shawnborton to confirm what is the expected look of this:

Current:

<img width="389" alt="Screenshot 2023-06-16 at 12 10 09 PM" src="https://github.com/Expensify/App/assets/87341702/946b4798-c641-49b2-8c28-9ac46268c98b">


This following is the expected background for the Share QR, right?

<img width="449" alt="Screenshot 2023-06-16 at 12 11 44 PM" src="https://github.com/Expensify/App/assets/87341702/2858b2d3-f614-46c6-8787-91a72b2469ae">

**Comment by shawnborton** [comment]
@0xmiroslav where did we end up with this? I think we originally decided that we should use the icon color as the BG color behind all transparent avatars everywhere in the app. However, I can see an argument to do something like the second screenshot above just for this particular QR code screen.

**Comment by aldo-expensify** [comment]
If we agree with going with the second option above, we just have to revert https://github.com/Expensify/App/pull/19883

**Comment by aldo-expensify** [comment]
> I think this ticket is not needed since we just have to revert https://github.com/Expensify/App/pull/19883.

Agreed, unless there is a payment for reporting for @alitoshmatov . @trjExpensify do you think reporting reward applies here?


> However, I don't think that the revert should be treated as a regression because as seen clearly in the conversation https://github.com/Expensify/App/issues/19569, the expectation was to set the same background colour as we're setting elsewhere.

I'm fine with not considering it a regression, I understand that there was work in parallel addressing similar things in separate issues/PRs made it pretty hard to see that this was going to happen. I'm not sure what it means for your compensation on https://github.com/Expensify/App/pull/19883 if we just revert it.

@allroundexperts can you put up a revert PR?

**Comment by aldo-expensify** [comment]
> #20938
> 
> Do I need a checklist for the revert as well @aldo-expensify?

Thanks, I added the checklist to the body.

**Comment by alitoshmatov** [comment]
The issue was that QR codes with expensify logo and non-transparent avatars were not checked after the changes. These changes were only aimed at fixing transparent avatars.

**Comment by trjExpensify** [comment]
>Agreed, unless there is a payment for reporting for @alitoshmatov . @trjExpensify do you think reporting reward applies here?

Was this the first time the bug was surfaced? If so, then I think payment for the bug report would be due.

**Comment by aldo-expensify** [comment]
> Was this the first time the bug was surfaced? If so, then I think payment for the bug report would be due.

I think it is the first time, I couldn't find other mentions.

**Comment by trjExpensify** [comment]
Offer sent for the bug report @alitoshmatov [here](https://www.upwork.com/jobs/~013a3d4105d7772fca).

**Comment by alitoshmatov** [comment]
Accepted offer
